test/grid_space_tests.jl

```diff
@@ -262,6 +262,135 @@ end
         @test a.pos == (2, 2, 2)
     end
 
+    @testset "random walk" begin
+        # random walks on grid spaces with euclidean metric are not defined
+        space = SpaceType((10,10), metric=:euclidean)
+        model = ABM(GridAgent2D, space)
+        add_agent!(model)
+        r = 1.0
+        @test_throws ArgumentError randomwalk!(model.agents[1], model, r)
+
+        # chebyshev metric
+        space = SpaceType((100,100), metric=:chebyshev)
+        model = ABM(GridAgent2D, space)
+        x₀ = (50,50)
+        add_agent!(x₀, model)
+        r = 1.5
+        randomwalk!(model.agents[1], model, r)
+        x₁ = model.agents[1].pos
+        # chebyshev distance after the random step should be 1
+        @test maximum(abs.(x₁ .- x₀)) == 1
+        # the list of available offsets at distance 1
+        # should be stored in space.offsets_at_radius[1]
+        @test haskey(space.offsets_at_radius, 1)
+        offsets_1 = Set((
+            (1,-1)  ,  (1,0)  ,  (1,1),
+            (0,-1)       ,       (0,1),
+            (-1,-1) ,  (-1,0) ,  (-1,1)
+        ))
```

Comment:
You should **not** test this. We only use the public API in the tests. Fields stored inside the spaces are not public api and hence not accessible / shouldn't be accessed. The tests here should be like in the continuous space by just testing the logical conclusions. (which you already do, so simply remove tests that don't use public api)